shapeshift/web
# feat: ens address lookups

This is a duplicate of https://github.com/shapeshift/web/pull/704 so cloudflare worker can run.

## Description

This adds client-side support for ENS resolution.
- Lookup (address -> domain)
- Reverse lookup (domain -> address)
- Fix inverted `from` and `to` addresses (dce23954395418cde1b80dbba230418465671492 by @0xApotheosis)

Refer to closing issue for full spec.

## Notice

Before submitting a pull request, please make sure you have answered the following:


## Pull Request Type


## Issue

Closes #674

## Testing

1. Try to pass a valid, resolvable ENS address as input in "Send" and complete flow to send confirm click
2. Metamask modal should appear with the ENS as destination (it might take a second or two to switch from 0x address to ENS name the first time, this is normal behavior)
3. Try to pass invalid ENS addresses e.g addresses not fitting the UTS46 spec or addresses with no owner/resolver should error. The flow I've decided to go with is to have "Next" button in loading state while checking for resolution, then erroring with some error message copy
4. Have some transactions for your current address with sender that has ENS domain resolving to it, it should be displayed instead of address
5. Click receive with a connected account that has ENS domain resolving to it, a rounded tag should be displayed simillar to acceptation criteria 

## Screenshots

![image](https://user-images.githubusercontent.com/17035424/148746188-58a79100-44f0-40f6-89fa-84a1281c2ac7.png)
![image](https://user-images.githubusercontent.com/17035424/148746222-a6b1b150-2ce6-4fb0-ae7a-46ff4ca461ec.png)
![image](https://user-images.githubusercontent.com/17035424/148746264-90510a0b-c00e-4708-b45f-41163877a4b7.png)
![image](https://user-images.githubusercontent.com/17035424/148746333-b8fe35c0-def0-4fca-8161-44d5364a5f09.png)
![image](https://user-images.githubusercontent.com/17035424/148857880-695aec09-74f6-4269-9509-647525c7f88a.png)
![image](https://user-images.githubusercontent.com/17035424/149047445-c3f3be24-0c37-4611-855e-d07427f13ae8.png)
